### PR TITLE
Add support for BigInt

### LINKED ISSUE #39: Support BigInt detection
http://2ality.com/2017/03/es-integer.html
https://groups.google.com/forum/#!msg/v8-dev/x571Gr0khNo/y8Jk0_vSBAAJ

Also `BigInt64Array` and `BigUint64Array`

Blocked by https://github.com/Microsoft/TypeScript/issues/15096

### PR DESCRIPTION
**Pull request by vladfrangu**
Fixes #39 

- [x] Implements support for the BigInt, BigInt64Array and BigUint64Array

- [x] All tests pass

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why are you removing Node.js 8 support? That seems unrelated.

**Comment by vladfrangu** [comment]
I thought nodejs 8 didn't have bigint support. Let me check real quick...

**Comment by vladfrangu** [comment]
image

nodejs 8 does NOT have BigInt support. (quite a shame too)

**Comment by sindresorhus** [comment]
We're not actually *using* BigInt, we're just detecting it, so we can still preserve support for Node.js 8. You can just comment out the tests for now.

File: tsconfig.json [hunk 1]
```diff
@@ -2,10 +2,10 @@
 	"extends": "@sindresorhus/tsconfig",
 	"compilerOptions": {
 		"outDir": "dist",
-		"target": "es2017",
+		"target": "esnext",
```
Comment (sindresorhus) [inline tsconfig.json]: We don't have to change `target`. TS doesn't polyfill BigInt anyway.

File: tsconfig.json [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 		"lib": [
 			"es2017",
```
Comment (sindresorhus) [inline tsconfig.json]: This can be removed now as `esnext` includes all previous versions.

**Comment by vladfrangu** [comment]
Yep, after reading your review, I realized you only did typeof / toString calls and checks, which would mean nodejs 8 support would still be present! I've done your requested changes now, as well as commented out all tests that only work on nodejs >=10. Hope this helps!

File: test/test.ts [hunk 3]
```diff
@@ -2,7 +2,6 @@ import fs from 'fs';
 import net from 'net';
 import Stream from 'stream';
 import util from 'util';
-import {URL} from 'url';
```
Comment (sindresorhus) [inline test/test.ts]: This too

File: .travis.yml [hunk 4]
```diff
@@ -1,4 +1,3 @@
 language: node_js
 node_js:
   - '10'
-  - '8'
```
Comment (sindresorhus) [inline .travis.yml]: And this

**Comment by vladfrangu** [comment]
@sindresorhus should be done now! If there's any other changes you'd like me to do, please let me know 👍

**Comment by sindresorhus** [comment]
Nice work 👌

**Pull request merged**